manticore/core/manticore.py

```diff
@@ -278,6 +283,69 @@ def _join_workers(self):
             while len(self._workers) > 0:
                 self._workers.pop().join()
 
+    #
+    # Symbol resolution
+    #
+
+    def _addr_by_symline(dwarfinfo, symbols, line):
+        '''
+        Return address for a symbol and line it is on through DWARF
+        debug info
+        '''
+        for symbol in symbols:
+            for CU in dwarfinfo.iter_CUs():
+                lineprog = dwarfinfo.line_program_for_CU(CU)
+                prevstate = None
+
+                for entry in lineprog.get_entries():
+                    # check for new state assignment
+                    if entry.state is None or entry.state.end_sequence:
+                        continue
+                    # check between address range between prev and current state containing address
+                    if prevstate and prevstate.address <= symbol['st_value'] < entry.state.address:
+                        state_line = prevstate.line
+                        if int(state_line) is line:
+                            return symbol['st_value']
+                    prevstate = entry.state
+        
+        return None, None
+
+
+    def resolve(self, symbol, line=None):
+        '''
+        A helper method used to resolve a symbol name into a memory address when
+        injecting hooks for analysis. 
+
+        :param symbol: function name to be resolved
+        :type symbol: string
+
+        :param line: if more functions present, optional line number can be included
+        :type line: int or None
+        '''
+        with open(self._path, 'rb') as f:
+
+            elffile = ELFFile(f)
+            if elffile is None:
+                return NotImplementedError("Symbols aren't supported")
+ 
+            if not elffile.has_dwarf_info():
+                raise NotImplementedError("DWARF debug info not supported")
+            dwarfinfo = elffile.get_dwarf_info()
+          
+            for section in elffile.iter_sections():
+                if not isinstance(section, SymbolTableSection):
+                    continue
+
+                symbols = section.get_symbol_by_name(symbol)
+                if not symbols:
+                    continue
+                
+                if line is None:
+                    return symbols[0].entry['st_value']
```

Comment:
Can there be a situation when there are many symbols with same name i.e. `len(symbols) > 1` is `True`?

What shall we do in such situation?

Reply:
Maybe we should raise an Exception and display info for all symbols to the user? So they can use `line=X` later on (and have the info about `line` from here).

Reply:
If there are many symbols that exist with the same name, the user can select the one that they would want to resolve through its line number. With this being an optional parameter, if `line` is not specified, `m.resolve` returns the address for the first instance of the symbol it discovers.

I think an exception could be used, but instead to be thrown if no symbols are not found _at all_.